src/operation.rs

```diff
@@ -473,6 +493,48 @@ impl Operation {
                     .send(Message::PendingProgress(id, 100.0))
                     .await;
             }
+            Self::Extract { paths, to } => {
+                for path in paths {
+                    let to = to.to_owned();
+
+                    tokio::task::spawn_blocking(move || -> Result<(), String> {
+                        if let Some(file_stem) = path.file_stem() {
+                            let mut new_dir = to.join(file_stem);
+                            if new_dir.exists() {
+                                let mut extensionless_path = path.to_owned();
+                                extensionless_path.set_extension("");
+                                if let Some(new_dir_parent) = new_dir.parent() {
+                                    new_dir = copy_unique_path(&extensionless_path, new_dir_parent);
+                                }
+                            }
+
+                            if let Some(mime) = mime_guess::from_path(&path).first() {
+                                match mime.essence_str() {
```

Comment:
The `zip` crate supports more: LZMA, BZIP2, ZSTD, but looks like w/o handling `tar`, at least there's no examples. Now I remember using other async crate that let me manipulate TAR archives with different compression format, but I don't have access to that code anymore (commercial thing).

I think it was [`async-compression`](https://crates.io/crates/async-compression).

Reply:
Hm, I tried enabling MIME types for those other formats with `zip`, and I couldn't get it to work for extracting those files. I did just add the `tar` crate so it supports uncompressed tar files now.